arch/sim/src/sim/up_hcisocket_host.c

```diff
@@ -0,0 +1,191 @@
+/****************************************************************************
+ * arch/sim/src/sim/up_hcisocket.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/types.h>
+#include <sys/uio.h>
+#include <sys/socket.h>
+#include <sys/ioctl.h>
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+
+#include "up_internal.h"
+#include "up_hcisocket_host.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define BTPROTO_HCI 1
+
+#define HCI_CHANNEL_RAW   0
+#define HCI_CHANNEL_USER  1
+
+#define HCIDEVDOWN  0x400448ca
+
+/* HCI data types */
+
+#define HCI_COMMAND_PKT   0x01
```

Comment:
Is it on purpose that these are redefined here? They were also defined in the non-host .c file

Reply:
Good catch I removed them.  The NuttX side is where the mapping of types happens so they are not needed on the host side.